src/swaps/bitcoin.rs

```diff
@@ -1163,35 +1168,68 @@ impl SwapScriptCommon for BtcSwapScript {
         keys: &Keypair,
         pub_nonce: &str,
         transaction_hash: &str,
-    ) -> Result<(MusigPartialSignature, MusigPubNonce), Error> {
+    ) -> Result<(musig::PartialSignature, musig::PublicNonce), Error> {
         // Step 1: Start with a Musig KeyAgg Cache
-        let secp = Secp256k1::new();
 
         let mut key_agg_cache = self.musig_keyagg_cache();
 
-        let tweak = SecretKey::from_slice(self.taproot_spendinfo()?.tap_tweak().as_byte_array())?;
+        let tweak = Scalar::from_be_bytes(*self.taproot_spendinfo()?.tap_tweak().as_byte_array())?;
 
-        let _ = key_agg_cache.pubkey_xonly_tweak_add(&secp, tweak)?;
+        let _ = key_agg_cache.pubkey_xonly_tweak_add(&tweak)?;
 
-        let session_id = MusigSessionId::new(&mut thread_rng());
+        let session_secret_rand = musig::SessionSecretRand::assume_unique_per_nonce_gen(rng_32b());
 
-        let msg = Message::from_digest_slice(&Vec::from_hex(transaction_hash)?)?;
+        let msg = hex_to_bytes32(transaction_hash)?;
 
         // Step 4: Start the Musig2 Signing session
         let mut extra_rand = [0u8; 32];
         OsRng.fill_bytes(&mut extra_rand);
 
-        let (gen_sec_nonce, gen_pub_nonce) =
-            key_agg_cache.nonce_gen(&secp, session_id, keys.public_key(), msg, Some(extra_rand))?;
+        let (gen_sec_nonce, gen_pub_nonce) = key_agg_cache.nonce_gen(
+            session_secret_rand,
+            convert_public_key(keys.public_key()),
+            &msg,
+            Some(extra_rand),
+        );
 
-        let boltz_nonce = MusigPubNonce::from_slice(&Vec::from_hex(pub_nonce)?)?;
+        let boltz_nonce = musig::PublicNonce::from_str(pub_nonce)?;
 
-        let agg_nonce = MusigAggNonce::new(&secp, &[boltz_nonce, gen_pub_nonce]);
+        let agg_nonce = musig::AggregatedNonce::new(&[&boltz_nonce, &gen_pub_nonce]);
 
-        let musig_session = MusigSession::new(&secp, &key_agg_cache, agg_nonce, msg);
+        let musig_session = musig::Session::new(&key_agg_cache, agg_nonce, &msg);
 
-        let partial_sig = musig_session.partial_sign(&secp, gen_sec_nonce, keys, &key_agg_cache)?;
+        let partial_sig =
+            musig_session.partial_sign(gen_sec_nonce, &convert_keypair(keys), &key_agg_cache);
 
         Ok((partial_sig, gen_pub_nonce))
     }
 }
+
+fn convert_pubkeys_for_musig(
+    pubkeys: &[bitcoin::secp256k1::PublicKey; 2],
+) -> [secp256k1_musig::PublicKey; 2] {
+    [
+        convert_public_key(pubkeys[0]),
+        convert_public_key(pubkeys[1]),
+    ]
+}
+
+fn convert_xonly_key(key: secp256k1_musig::XOnlyPublicKey) -> bitcoin::XOnlyPublicKey {
+    bitcoin::XOnlyPublicKey::from_slice(&key.serialize()[..]).expect("xonly key size matches")
```

Comment:
these should basically never error, right?

Reply:
...all the convert* methods

Reply:
Yes, those are basically the same types from different versions of the library which of course maintained serialization compatibility.
Once the last release of secp is included in bitcoin/elements crate we can avoid this